pkg/kubernetes/diff.go

```diff
@@ -209,3 +217,13 @@ func (m multiDiff) diff() (*string, error) {
 	}
 	return &diff, nil
 }
+
+func findNamespaces(state manifest.List) map[string]bool {
+	namespaces := map[string]bool{}
+	for _, m := range state {
+		if namespace := m.Metadata().Namespace(); namespace != "" {
```

Comment:
We do heavy validation of the resource we operate on here, namespace is guaranteed to be set, so checking it here is technically redundant

Reply:
guaranteed? What about cluster-wide resources?

Reply:
good point, I did not think of this